### ISSUE TITLE
Crash while editing surfaces

### ISSUE THREAD

**Issue by cpicanco** [pupil-labs/pupil#107]
- Select the edit surface mode
- Add a new surface
- Try to edit the surface

```bash
OpenCV Error: Assertion failed (scn + 1 == m.cols && (depth == CV_32F || depth == CV_64F)) in perspectiveTransform, file /build/buildd/opencv-2.4.8+dfsg1/modules/core/src/matmul.cpp, line 1936
Traceback (most recent call last):
  File "/home/rafael/cpicanco/pupil/pupil_src/player/main.py", line 472, in <module>
    main()
  File "/home/rafael/cpicanco/pupil/pupil_src/player/main.py", line 415, in main
    p.update(frame,events)
  File "/home/rafael/cpicanco/pupil/pupil_src/shared_modules/offline_marker_detector.py", line 257, in update
    s.move_vertex(v_idx,new_pos)
  File "/home/rafael/cpicanco/pupil/pupil_src/shared_modules/reference_surface.py", line 335, in move_vertex
    m.uv_coords = cv2.perspectiveTransform(m.uv_coords,transform)
cv2.error: /build/buildd/opencv-2.4.8+dfsg1/modules/core/src/matmul.cpp:1936: error: (-215) scn + 1 == m.cols && (depth == CV_32F || depth == CV_64F) in function perspectiveTransform
```

**Comment by willpatera** [comment]
@cpicanco -- I am unable to recreate your error. Could you describe further so I might be able to reproduce the error?

**Comment by cpicanco** [comment]
I could not replicate this issue with the last bundle, only running from source from my local repository and a fresh cloned one.

So with the last commit.

- I am testing with a recording that have a single marker.
- I have closed all plugins, except vis_circle and offline_marker_detector.
- The crash occur while trying to edit a second surface by clicking on a vertice.

**Comment by willpatera** [comment]
Ok - I will try to replicate this and get back to you. Thanks for the update.

**Comment by willpatera** [comment]
Ok - I was able to replicate your error. I'm looking into it now.

**Comment by willpatera** [comment]
I made a quick fix so that Player won't crash here: https://github.com/pupil-labs/pupil/commit/8bd1704919e8b922294849cd0c9cf668adef18c1 -- and will close this issue. 

It appears that when you add a surface it takes a little bit of time to add a new marker and compute the `uv_coordinates` -- (a very small amount of time - quick test shows around 15 `None` outputs in the terminal from print, before the `uv_coords` value is available). And when you are already in edit mode and then quickly add a marker and then try to edit right away the marker's u_v coordinates will be `None` for this short period of time and will lead to the crash when OpenCV tries to compute the perspective transform.

I just added one line of code as a check to make sure we actually have the coordinates before we try to use them.
